src/app/common/formly/types/master-details/master-details.ts

```diff
@@ -0,0 +1,178 @@
+import index from '@angular/cli/lib/cli';
+import { Component, OnInit } from '@angular/core';
+import { FormArray, FormBuilder, FormGroup } from '@angular/forms';
+import { FieldType, FormlyFieldConfig, FormlyFormBuilder } from '@ngx-formly/core';
+import * as clonedeep from 'lodash.clonedeep';
+
+@Component({
+	selector: 'formly-mater-details',
+	template: `
+		<div class="row">
+			<div class="col-4">
+
+				<div class="card form-group">
+					<div class="card-header">
+						<span *ngFor="let column of columns">{{column.templateOptions.label}}</span>
+						<span *ngIf="autogeneratedTitleColumn">{{to['titleColumnLabel']}}</span>
+						<button
+							style="position: absolute; right: 15px;"
+							type="button" mat-icon-button [disabled]="detailForm && detailForm.invalid" (click)="add(true)">
+							<mat-icon>add</mat-icon>
+						</button>
+					</div>
+					<div class="card-block">
+						<table class="table">
+							<tbody>
+								<tr *ngFor="let item of model; let i = index;"
+									[ngClass]="{ 'table-active': currentIndex == i }"
+									(click)="selectItem(i)"
+								>
+									<td *ngFor="let column of columns" [ngStyle]="{ 'text-decoration': isRemoved(i) ? 'line-through' : 'none' }">{{item[column.key] || '[NEW]'}}</td>
+									<td *ngIf="autogeneratedTitleColumn" [ngStyle]="{ 'text-decoration': isRemoved(i) ? 'line-through' : 'none' }">
+										{{to.titleColumnValueGetter(item) || '[NEW]'}}
+									</td>
+									<td width="40px">
+										<button type="button" color="warn" mat-icon-button (click)="remove(i)" *ngIf="!isRemoved(i)">
+											<mat-icon>delete_forever</mat-icon>
+										</button>
+										<button type="button" mat-icon-button (click)="remove(i)" *ngIf="isRemoved(i)">
+											<mat-icon>undo</mat-icon>
+										</button>
+									</td>
+								</tr>
+							</tbody>
+						</table>
+					</div>
+				</div>
+
+			</div>
+			<div class="col-8" *ngIf="currentItem">
+
+				<div class="card form-group">
+					<div class="card-header">
+						Citation
+					</div>
+					<div class="card-block" style="padding: 20px;">
+						<form [formGroup]="detailForm" *ngIf="currentItem">
+							<formly-form
+								[form]="detailForm"
+								[fields]="fields[currentIndex]"
+								[model]="model[currentIndex]"
+								[options]="options">
+							</formly-form>
+							<div class="stub" *ngIf="isRemoved(currentIndex)"></div>
+						</form>
+					</div>
+				</div>
+
+			</div>
+		</div>
+	`,
+	styleUrls: ['./master-details.scss']
+})
+export class FormlyMasterDetails extends FieldType implements OnInit {
```

Comment:
you can use `FieldArrayType` instead, see https://formly-js.github.io/ngx-formly/examples/advanced/repeating-section [repeat-section.type](https://raw.githubusercontent.com/formly-js/ngx-formly/master/demo/src/app/examples/advanced/repeating-section/repeat-section.type.ts)